### PR TITLE
fix: Implement atomic mode OBJECTS to commit only valid DTOs

### PR DESCRIPTION
**Pull request by enricocolasante**

### CODE REVIEW

**Review by stian-sandvold — APPROVED** [review]
Looks good! But same codestyle issues as in the previous PR :)

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/report/TrackerValidationReport.java [hunk 1]
```diff
@@ -57,6 +58,11 @@
     @JsonProperty
     @Builder.Default
     private List<TrackerValidationHookTimerReport> performanceReport = new ArrayList<>();
+
+    @JsonProperty
+    @Builder.Default
+    private List<TrackerDto> notValidDtos = new ArrayList<>();
```
Comment (ameenhere) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/report/TrackerValidationReport.java:64]: is "invalidDtos" a better name? or maybe there is a reason for the naming. Just curious.

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/params/AtomicModeIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.hisp.dhis.tracker.params;
```
Comment (ameenhere) [inline dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/params/AtomicModeIntegrationTest.java:29]: package before copyright

**Review by ameenhere — APPROVED** [review]
Minor changes/suggestions

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/report/TrackerValidationReport.java (hunk 1 above)
Comment (enricocolasante) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/report/TrackerValidationReport.java]: It was probably validDtos and then I added the not.
InvalidDtos makes much more sense :)

**Pull request merged**